runtime/drive_handler.go

```diff
@@ -0,0 +1,192 @@
+// Copyright 2018-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sync"
+	"sync/atomic"
+
+	"github.com/pkg/errors"
+
+	firecracker "github.com/firecracker-microvm/firecracker-go-sdk"
+	models "github.com/firecracker-microvm/firecracker-go-sdk/client/models"
+
+	"github.com/firecracker-microvm/firecracker-containerd/internal"
+)
+
+const (
+	minFCDriveImageBytes = 512
+)
+
+var (
+	// ErrStubDrivesExhausted occurs when there are no more stub drives left to
+	// use. This can happen by calling PatchStubDrive greater than the stub drive
+	// list.
+	ErrStubDrivesExhausted = fmt.Errorf("There are no remaining stub drives to use")
+
+	// ErrDriveIDNil should never happen, but we safe guard against nil dereferencing
+	ErrDriveIDNil = fmt.Errorf("DriveID of current drive is nil")
+
+	// ErrVMNotFound means that a VM with the given ID not found
+	ErrVMNotFound = errors.New("vm not found")
+)
+
+// stubDriveHandler is used to create a pool of stubbed file. This will allow
+// different VMs to use the same files and if a VM requires more files than
+// provided, this will handle increasing the files.
+type stubDriveHandler struct {
+	RootPath string
+
+	drives sync.Map
```

Comment:
Do we still need this map if we're not sharing the struct among multiple VMs?  I assume we just need a single `driveManager` struct and its `[]models.Drive`.